pkg/storage/storagewrappers/boundedconcurrency.go

```diff
@@ -0,0 +1,86 @@
+package storagewrappers
+
+import (
+	"context"
+	"time"
+
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const TimeWaitingSpanAttribute = "time_waiting"
+
+var _ storage.RelationshipTupleReader = (*boundedConcurrencyTupleReader)(nil)
+
+var (
+	timeWaitingCounter = promauto.NewHistogram(prometheus.HistogramOpts{
+		Name:    "time_waiting_for_bounded_query",
```

Comment:
It's standard practice to name the metric with a suffix indicating the units. 

Also, minor nit, it would be nice to rename this to `datastore_bounded_concurrency_lag_ms` or `datastore_bounded_concurrency_delay_ms` or `datastore_read_limiter_delay_ms` or something similar. `time_waiting_for_bounded_query` isn't clear to me.